File: Code/MyApplication7/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.PopupMenu;

public class MainActivity extends AppCompatActivity {

    // 声明控件
    private Button btnRightNow;
    private Button btnMaybeLater;
    private ImageButton btnPopupMenu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 绑定控件
        btnRightNow = findViewById(R.id.btnRightNow);
        btnMaybeLater = findViewById(R.id.btnMaybeLater);
        btnPopupMenu = findViewById(R.id.btnPopupMenu);

        // "+"按钮点击事件（弹出菜单）
        btnPopupMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                PopupMenu popupMenu = new PopupMenu(MainActivity.this, v);
                popupMenu.getMenuInflater().inflate(R.menu.popup_menu, popupMenu.getMenu());
                popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
                    @Override
                    public boolean onMenuItemClick(MenuItem item) {
                        // 将switch语句改为if-else语句以解决"Constant expression required"问题
                        if (item.getItemId() == R.id.menu_new_chat) {
                            Toast.makeText(MainActivity.this, "New Chat", Toast.LENGTH_SHORT).show();
                            return true;
                        } else if (item.getItemId() == R.id.menu_add_contacts) {
                            Toast.makeText(MainActivity.this, "Add Contacts", Toast.LENGTH_SHORT).show();
                            return true;
                        } else if (item.getItemId() == R.id.menu_scan) {
                            Toast.makeText(MainActivity.this, "Scan", Toast.LENGTH_SHORT).show();
                            return true;
                        } else if (item.getItemId() == R.id.menu_money) {
                            Toast.makeText(MainActivity.this, "Money", Toast.LENGTH_SHORT).show();
                            return true;
                        } else if (item.getItemId() == R.id.menu_support) {
                            Toast.makeText(MainActivity.this, "Support", Toast.LENGTH_SHORT).show();
                            return true;
                        } else {
                            return false;
                        }
                    }
                });
                popupMenu.show();
            }
        });

        // RIGHT NOW 按钮点击事件
        btnRightNow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "RIGHT NOW", Toast.LENGTH_SHORT).show();
            }
        });

        // MAYBE LATER 按钮点击事件
        btnMaybeLater.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "MAYBE LATER", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: 11637 MAPE 2 1025/Code/MyApplication10/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.activity.OnBackPressedCallback
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
    private lateinit var drawerLayout: CustomDrawerLayout
    private lateinit var navigationView: NavigationView
    private var closeMenuItem: MenuItem? = null
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_menu)
        
        drawerLayout = findViewById(R.id.drawer_layout)
        navigationView = findViewById(R.id.nav_view)
        navigationView.setNavigationItemSelectedListener(this)
        
        // 监听抽屉状态变化
        drawerLayout.addDrawerListener(object : DrawerLayout.DrawerListener {
            override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
                // 抽屉滑动时的处理
            }
            
            override fun onDrawerOpened(drawerView: View) {
                // 抽屉打开时显示关闭按钮
                closeMenuItem?.isVisible = true
                supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_close)
            }
            
            override fun onDrawerClosed(drawerView: View) {
                // 抽屉关闭时隐藏关闭按钮
                closeMenuItem?.isVisible = false
                supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_menu)
            }
            
            override fun onDrawerStateChanged(newState: Int) {
                // 抽屉状态改变时的处理
            }
        })
        
        // 处理返回键事件
        onBackPressedDispatcher.addCallback(this, object : OnBackPressedCallback(true) {
            override fun handleOnBackPressed() {
                if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
                    drawerLayout.closeDrawer(GravityCompat.START)
                } else {
                    isEnabled = false
                    onBackPressedDispatcher.onBackPressed()
                }
            }
        })
        
        // Load default fragment
        if (savedInstanceState == null) {
            loadFragment(HomeFragment())
            navigationView.setCheckedItem(R.id.nav_home)
        }
    }
    
    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        val fragment: Fragment = when (item.itemId) {
            R.id.nav_home -> {
                Toast.makeText(this, "首页", Toast.LENGTH_SHORT).show()
                HomeFragment()
            }
            R.id.nav_profile -> {
                Toast.makeText(this, "个人资料", Toast.LENGTH_SHORT).show()
                ProfileFragment()
            }
            R.id.nav_settings -> {
                Toast.makeText(this, "设置", Toast.LENGTH_SHORT).show()
                SettingsFragment()
            }
            R.id.nav_qr_code -> {
                Toast.makeText(this, "二维码", Toast.LENGTH_SHORT).show()
                QrCodeFragment()
            }
            else -> HomeFragment()
        }
        
        loadFragment(fragment)
        drawerLayout.closeDrawer(GravityCompat.START)
        return true
    }
    
    // 添加菜单
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.toolbar_menu, menu)
        closeMenuItem = menu.findItem(R.id.action_close)
        closeMenuItem?.isVisible = false // 默认隐藏关闭按钮
        return true
    }
    
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
                    drawerLayout.closeDrawer(GravityCompat.START)
                } else {
                    drawerLayout.openDrawer(GravityCompat.START)
                }
                true
            }
            R.id.action_close -> {
                // 关闭菜单并返回主页面（首页）
                drawerLayout.closeDrawer(GravityCompat.START)
                loadFragment(HomeFragment())
                navigationView.setCheckedItem(R.id.nav_home)
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
    
    private fun loadFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, fragment)
            .commit()
    }
}